cloud_governance/cloud_resource_orchestration/aws/ec2/cost_over_usage.py

```diff
@@ -0,0 +1,167 @@
+                                    "timestamp": {
+                                        "format": "yyyy-MM-dd",
+                                        "lte": str(self.current_end_date.date()),
+                                        "gte": str(self.current_start_date.date()),
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                response = self.es_operations.fetch_data_by_es_query(es_index=self.es_index_cro, query=query)
+                if not response:
+                    over_usage_users.append(user)
+                for cro_data in response:
+                    project = cro_data.get('_source').get('project')
+                    ticket_id_state = cro_data.get('_source').get('ticket_id_state')
+                    if 'close' in ticket_id_state.lower():
+                        ticket_closed_time = datetime.strptime(cro_data.get('_source').get('timestamp'), "%Y-%m-%dT%H:%M:%S.%f")
+                        month_start_date = ticket_closed_time if self.current_start_date < ticket_closed_time else self.current_start_date
+                        user_response = self.get_monthly_user_es_cost_data(start_date=month_start_date, extra_matches={'Tags': {'Key': 'User', 'Values': [user_name]}}, extra_operation='And', tag_name='Project')
+                        if user_response:
+                            if project in [res.get('Project') for res in user_response]:
+                                user_cost = round(user_response[0].get('Cost'), self.DEFAULT_ROUND_DIGITS)
+                                user['Cost'] = user_cost
+                                user['Project'] = project
+                                if user_cost >= self.__over_usage_amount:
+                                    over_usage_users.append(user)
+        return over_usage_users
+
+    def send_alerts_to_over_usage_users(self):
+        """
+        This method send alerts to cost over usage users
+        :return:
+        """
+        users_list = self.get_cost_over_usage_users()
+        for row in users_list:
+            user, cost, project = row.get('User'), row.get('Cost'), row.get('Project', '')
+            cc = [*self.__cro_admins]
+            user_details = self.__ldap_search.get_user_details(user_name=user)
+            if user_details:
+                name = f'{user_details.get("FullName")}'
+                cc.append(user_details.get('managerId'))
+                subject, body = self.__mail_message.get_cost_over_usage(CloudName=self.__public_cloud_name,
+                                                                        OverUsageCost=self.__over_usage_amount,
+                                                                        FullName=name, Cost=cost, Project=project)
+                self.__postfix_mail.send_email_postfix(to=user, cc=cc, content=body, subject=subject, mime_type='html')
```

Comment:
I think we should method send_email with 2 options: postfix or private gmail, that will give flexibility to run it also in public, wdyt ?

Reply:
yes, Good.